recipe/build.sh

```diff
@@ -13,39 +13,8 @@ CMAKE_FLAGS+=" -DTorch_DIR=${SP_DIR}/torch/share/cmake/Torch"
 # adapted from https://github.com/conda-forge/faiss-split-feedstock/blob/main/recipe/build-lib.sh
 declare -a CUDA_CONFIG_ARGS
 if [ ${cuda_compiler_version} != "None" ]; then
-    # the following are all the x86-relevant gpu arches; for building aarch64-packages, add: 53, 62, 72
-    ARCHES=(52 60 61 70)
-    # cuda 11.0 deprecates arches 35, 50
-    DEPRECATED_IN_11=(35 50)
-    if [ $(version2int $cuda_compiler_version) -ge $(version2int "11.1") ]; then
-        # Ampere support for GeForce 30 (sm_86) needs cuda >= 11.1
-        LATEST_ARCH=86
-        # ARCHES does not contain LATEST_ARCH; see usage below
-        ARCHES=( "${ARCHES[@]}" 75 80 )
-    elif [ $(version2int $cuda_compiler_version) -ge $(version2int "11.0") ]; then
-        # Ampere support for A100 (sm_80) needs cuda >= 11.0
-        LATEST_ARCH=80
-        ARCHES=( "${ARCHES[@]}" 75 )
-    elif [ $(version2int $cuda_compiler_version) -ge $(version2int "10.0") ]; then
-        # Turing support (sm_75) needs cuda >= 10.0
-        LATEST_ARCH=75
-        ARCHES=( "${DEPRECATED_IN_11[@]}" "${ARCHES[@]}" )
-    fi
-    for arch in "${ARCHES[@]}"; do
-        CMAKE_CUDA_ARCHS="${CMAKE_CUDA_ARCHS+${CMAKE_CUDA_ARCHS};}${arch}"
-    done
-    # for -real vs. -virtual, see cmake.org/cmake/help/latest/prop_tgt/CUDA_ARCHITECTURES.html
-    # this is to support PTX JIT compilation; see first link above or cf.
-    # devblogs.nvidia.com/cuda-pro-tip-understand-fat-binaries-jit-caching
-    CMAKE_CUDA_ARCHS="${CMAKE_CUDA_ARCHS};${LATEST_ARCH}-real;${LATEST_ARCH}-virtual"
-
-    CUDA_CONFIG_ARGS+=(
-        -DCMAKE_CUDA_ARCHITECTURES="${CMAKE_CUDA_ARCHS}"
-    )
-    ## cmake does not generate output for the call below; echo some info
-    echo "Set up extra cmake-args: CUDA_CONFIG_ARGS=${CUDA_CONFIG_ARGS+"${CUDA_CONFIG_ARGS[@]}"}"
-
-    CMAKE_FLAGS+=" ${CUDA_CONFIG_ARGS}"
+    ARCH_LIST=$(${PYTHON} -c "import torch; print(';'.join([arch.split('_')[1][:1] + '.' + arch.split('_')[1][1:] for arch in torch._C._cuda_getArchFlags().split()]))")
```

Comment:
Why use a private method rather than `torch.cuda.get_arch_list()`? AFAICT we should be able to use the output from the latter directly. Also, a bit of documentation would help.
```suggestion
    # sample output for torch.cuda.get_arch_list(): ['sm_37', 'sm_50', 'sm_60', 'sm_70', 'sm_75', 'sm_80', 'sm_86', 'sm_90']
    # we need to turn this into: "3.7;5.0;6.0;7.0;7.5;8.0;8.6;9.0"
    ARCH_LIST=$(python -c "import torch; print(';'.join([f'{y[0]}.{y[1]}' for y in [x[3:] for x in torch.cuda.get_arch_list()]]))")
```

Reply:
The higher level cuda.get_arch_list is currently defined like this:
```python
def get_arch_list() -> List[str]:
    r"""Returns list CUDA architectures this library was compiled for."""
    if not is_available():
        return []
    arch_flags = torch._C._cuda_getArchFlags()
    if arch_flags is None:
        return []
    return arch_flags.split()
```
It will return nothing if there is no GPU (which I think is a mistake, tbh).
I tried that to start with, but found out it did not work. Luckily the _C call is compliant always :P

Reply:
OK, fair enough. So then my suggestion would be
```suggestion
    # sample output for torch._C._cuda_getArchFlags(): "sm_37 sm_50 sm_60 sm_70 sm_75 sm_80 sm_86 sm_90"
    # we need to turn this into: "3.7;5.0;6.0;7.0;7.5;8.0;8.6;9.0"
    ARCH_LIST=$(python -c "import torch; print(';'.join([f'{y[0]}.{y[1]}' for y in [x[3:] for x in torch._C._cuda_getArchFlags().split()]]))")
```

Reply:
What about this then?
```python
python -c "import torch; print(';'.join([f'{y[0]}.{y[1]}' for y in [x[3:] for x in torch._C._cuda_getArchFlags().split() if x.startswith('sm_')]]))"
```

Reply:
Ah, yes, good point. I had found a sample output on the pytorch issue tracker, and it didn't have the `compute_` part.

That bit with the last arch being repeated at the end is actually relevant for the PTX thing; I'm not sure how TORCH_CUDA_ARCH does that behind the scenes (I guess it does the same). But your suggestion is fine, as long as you also update the comment above w.r.t the expected output of `_cuda_getArchFlags`.